### PR TITLE
Structure_Engine: Replace calls to Clone, IClone, and GetShallowClone methods with calls to DeepClone and ShallowClone 

### LINKED ISSUE #2201: Structure_Engine: Replace calls to Clone methods with calls to Base DeepClone or ShallowClone
#### Broken rules:
See https://github.com/BHoM/admin/issues/9

#### Suggestions to restore compliance:
https://github.com/BHoM/admin/issues/9#issuecomment-709032648

### PR DESCRIPTION
**Pull request by alelom**
### Issues addressed by this PR

Closes #2201

See BHoM/admin#9

- Replaced calls to Geometry.Clone and Geometry.IClone to calls to Engine.Base.DeepClone(). Feel free to use Engine.Base.ShallowClone() where you feel appropriate.
- Replaced calls to GetShallowClone with calls to Engine.Base.ShallowClone()

### Test files

Just compliance, no functional change.

- Review the code itself to make sure it compiles and makes sense
- Run the standard testing procedure that involve the structure engine

### CODE REVIEW

**Review by peterjamesnugent — DISMISSED** [review]
LGTM.

**Comment by alelom** [comment]
Thanks @peterjamesnugent , I realized I missed to replace a call to IClone.

**Review by peterjamesnugent — DISMISSED** [review]
LGTM - did a quick search and there's no more uses of `IClone` or `GetShallowClone` in Structure_Engine.

**Comment by alelom** [comment]
@BHoMBot check required
@BHoMBot check dataset-compliance
@BHoMBot check copyright-compliance
@BHoMBot check ready-to-merge

File: Structure_Engine/Modify/Transform.cs [hunk 1]
```diff
@@ -50,7 +51,7 @@ public static Node Transform(this Node node, TransformMatrix transform, double t
                 return null;
             }
 
-            Node result = node.GetShallowClone() as Node;
+            Node result = node.ShallowClone() as Node;
```
Comment (FraserGreenroyd) [inline Structure_Engine/Modify/Transform.cs:54]:
```suggestion
            Node result = node.ShallowClone();
```

File: Structure_Engine/Modify/Transform.cs [hunk 2]
```diff
@@ -71,7 +72,7 @@ public static Bar Transform(this Bar bar, TransformMatrix transform, double tole
                 return null;
             }
 
-            Bar result = bar.GetShallowClone() as Bar;
+            Bar result = bar.ShallowClone() as Bar;
```
Comment (FraserGreenroyd) [inline Structure_Engine/Modify/Transform.cs:75]:
```suggestion
            Bar result = bar.ShallowClone();
```

File: Structure_Engine/Modify/Transform.cs [hunk 3]
```diff
@@ -97,7 +98,7 @@ public static RigidLink Transform(this RigidLink link, TransformMatrix transform
                 return null;
             }
 
-            RigidLink result = link.GetShallowClone() as RigidLink;
+            RigidLink result = link.ShallowClone() as RigidLink;
```
Comment (FraserGreenroyd) [inline Structure_Engine/Modify/Transform.cs:101]:
```suggestion
            RigidLink result = link.ShallowClone();
```

File: Structure_Engine/Modify/Transform.cs [hunk 4]
```diff
@@ -118,7 +119,7 @@ public static Edge Transform(this Edge edge, TransformMatrix transform, double t
                 return null;
             }
 
-            Edge result = edge.GetShallowClone() as Edge;
+            Edge result = edge.ShallowClone() as Edge;
```
Comment (FraserGreenroyd) [inline Structure_Engine/Modify/Transform.cs:122]:
```suggestion
            Edge result = edge.ShallowClone();
```

File: Structure_Engine/Modify/Transform.cs [hunk 5]
```diff
@@ -138,7 +139,7 @@ public static Opening Transform(this Opening opening, TransformMatrix transform,
                 return null;
             }
 
-            Opening result = opening.GetShallowClone() as Opening;
+            Opening result = opening.ShallowClone() as Opening;
```
Comment (FraserGreenroyd) [inline Structure_Engine/Modify/Transform.cs:142]:
```suggestion
            Opening result = opening.ShallowClone();
```

File: Structure_Engine/Modify/Transform.cs [hunk 6]
```diff
@@ -158,7 +159,7 @@ public static Panel Transform(this Panel panel, TransformMatrix transform, doubl
                 return null;
             }
 
-            Panel result = panel.GetShallowClone() as Panel;
+            Panel result = panel.ShallowClone() as Panel;
```
Comment (FraserGreenroyd) [inline Structure_Engine/Modify/Transform.cs:162]:
```suggestion
            Panel result = panel.ShallowClone();
```

File: Structure_Engine/Modify/Transform.cs [hunk 7]
```diff
@@ -186,7 +187,7 @@ public static FEMesh Transform(this FEMesh mesh, TransformMatrix transform, doub
                 return null;
             }
 
-            FEMesh result = mesh.GetShallowClone() as FEMesh;
+            FEMesh result = mesh.ShallowClone() as FEMesh;
```
Comment (FraserGreenroyd) [inline Structure_Engine/Modify/Transform.cs:190]:
```suggestion
            FEMesh result = mesh.ShallowClone();
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
This PR has the same files changed as #2552 
I am prioritising #2552 for merge because it has already started the checks.

@alelom can you confirm the state of Facades_Engine as a result?

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required
@BHoMBot check dataset-compliance
@BHoMBot check copyright-compliance
@BHoMBot check ready-to-merge

**Review by peterjamesnugent — APPROVED** [review]
LGTM.

**Pull request merged**